ui/app/routes/vault/cluster/secrets/backend/configuration/edit.ts

```diff
@@ -0,0 +1,120 @@
+import { set } from '@ember/object';
+import Route from '@ember/routing/route';
+import { service } from '@ember/service';
+import { CONFIGURABLE_SECRET_ENGINES } from 'vault/helpers/mountable-secret-engines';
+import errorMessage from 'vault/utils/error-message';
+import { action } from '@ember/object';
+
+import type Store from '@ember-data/store';
+import type SecretEngineModel from 'vault/models/secret-engine';
+
+// This route file is reused for all configurable secret engines.
+// It generates config models based on the engine type.
+// Saving and updating of those models are done within the engine specific components.
+
+const CONFIG_ADAPTERS_PATHS: Record<string, string[]> = {
+  // aws: ['aws/lease-config', 'aws/root-config'], TODO will be uncommented when AWS refactor occurs
+  ssh: ['ssh/ca-config'],
+};
+
+export default class SecretsBackendConfigurationEdit extends Route {
+  @service declare readonly store: Store;
+
+  async model() {
+    const { backend } = this.paramsFor('vault.cluster.secrets.backend');
+    const secretEngineRecord = this.modelFor('vault.cluster.secrets.backend') as { type: SecretEngineModel };
+    const type = secretEngineRecord.type as string;
+
+    // if the engine type is not configurable, return a 404.
+    if (!secretEngineRecord || !CONFIGURABLE_SECRET_ENGINES.includes(type)) {
+      const error = new AdapterError();
+      set(error, 'httpStatus', 404);
+      throw error;
+    }
+    // TODO this conditional will be removed when we handle AWS
+    if (type !== 'aws') {
+      // generate the model based on the engine type.
+      // and pre-set with the type and backend (e.g. type: ssh, id: ssh-123)
+      const model: Record<string, unknown> = { type, id: backend };
+      for (const adapterPath of CONFIG_ADAPTERS_PATHS[type] as string[]) {
+        try {
+          model[adapterPath] = await this.store.queryRecord(adapterPath, {
+            backend,
+            type,
+          });
+        } catch (e: AdapterError) {
+          // For most models if the adapter returns a 404, we want to create a new record.
+          // The ssh secret engine however returns a 400 if the CA is not configured.
+          // For ssh's 400 error, we want to create the CA config model.
+          if (
+            e.httpStatus === 404 ||
+            (type === 'ssh' && e.httpStatus === 400 && errorMessage(e) === `keys haven't been configured yet`)
+          ) {
+            model[adapterPath] = await this.store.createRecord(adapterPath, {
+              backend,
+              type,
+            });
+          } else {
+            throw e;
+          }
+        }
+      }
+      // convert the adapterPath with a name that can be passed to the components
+      // ex: adapterPath = ssh/ca-config, convert to: ssh-ca-config so that you can pass to component @model={{this.model.ssh-ca-config}}
+      for (const key in model) {
+        this.standardizeModelName(key, model);
```

Comment:
instead of doing this afterward, I would suggest we standardize the name when we set it on the model at line 47, so we minimize methods with side effects